packages/cozy-client/src/CozyClient.js

```diff
@@ -477,20 +478,26 @@ export default class CozyClient {
   }
 
   async authorize(openURLCallback) {
-    const client = this.getClient()
-    const stateCode = client.generateStateCode()
-    const url = client.getAuthCodeURL(stateCode)
-
-    const redirectedURL = await openURLCallback(url)
-    const code = client.getAccessCodeFromURL(redirectedURL, stateCode)
-    const token = await client.fetchAccessToken(code)
-
-    client.setCredentials(token)
-
-    return {
-      token,
-      infos: client.oauthOptions,
-      client: client.oauthOptions // for compat with Authentication comp reasons
+    try {
+      const client = this.getClient()
+      const stateCode = client.generateStateCode()
+      const url = client.getAuthCodeURL(stateCode)
+      const redirectedURL = await openURLCallback(url)
+      const code = client.getAccessCodeFromURL(redirectedURL, stateCode)
+      const token = await client.fetchAccessToken(code)
+
+      client.setCredentials(token)
+      return {
+        token,
+        infos: client.oauthOptions,
+        client: client.oauthOptions // for compat with Authentication comp reasons
+      }
+    } catch (error) {
+      /* if REGISTRATION_ABORT is emited, we have to unregister the client. */
+      if (error.message === REGISTRATION_ABORT) {
+        const client = this.getClient()
+        client.unregister()
+      }
```

Comment:
Should we throw the error after the unregistration? If the app needs to do something itself this could be useful, and it wouldn't change the existing behavior (in Drive we'd to a catch with presumably nothing in it?)

Reply:
Done

Reply:
Actually I was thinking about all error cases, not just the abort, sorry if that wasn't very clear.
If there is an error in any of the other function calls, it will be silently absorbed here and we'll have to way to detect it.

Reply:
Good catch! Is it ok now?